### DISCUSSION TITLE
Compiling Noah-MP following tutorial

### DISCUSSION THREAD

**Discussion by psbecker** [NCAR/hrldas#190] [General]
I am trying to compile Noah-MP following the [tutorial](https://github.com/NCAR/hrldas/blob/develop/tutorial/Note0_Download_Compile.ipynb) but get the following error: 
` make
(cd Utility_routines;           make)
make[1]: Entering directory '/home/pabecker/hrldas/hrldas/Utility_routines'
make[1]: Nothing to be done for 'all'.
make[1]: Leaving directory '/home/pabecker/hrldas/hrldas/Utility_routines'
(cd ../noahmp/utility;          make)
make[1]: Entering directory '/home/pabecker/hrldas/noahmp/utility'
make[1]: Nothing to be done for 'all'.
make[1]: Leaving directory '/home/pabecker/hrldas/noahmp/utility'
(cd ../noahmp/src;              make)
make[1]: Entering directory '/home/pabecker/hrldas/noahmp/src'
make[1]: Nothing to be done for 'all'.
make[1]: Leaving directory '/home/pabecker/hrldas/noahmp/src'
(cd ../noahmp/drivers/hrldas;   make)
make[1]: Entering directory '/home/pabecker/hrldas/noahmp/drivers/hrldas'
make[1]: Nothing to be done for 'all'.
make[1]: Leaving directory '/home/pabecker/hrldas/noahmp/drivers/hrldas'
(cd ../urban/wrf;               make)
make[1]: Entering directory '/home/pabecker/hrldas/urban/wrf'
make[1]: Nothing to be done for 'all'.
make[1]: Leaving directory '/home/pabecker/hrldas/urban/wrf'
(cd IO_code;                    make)
make[1]: Entering directory '/home/pabecker/hrldas/hrldas/IO_code'
make[1]: Nothing to be done for 'all'.
make[1]: Leaving directory '/home/pabecker/hrldas/hrldas/IO_code'
(cd run;                        make)
make[1]: Entering directory '/home/pabecker/hrldas/hrldas/run'
make[1]: Nothing to be done for 'all'.
make[1]: Leaving directory '/home/pabecker/hrldas/hrldas/run'
(cd HRLDAS_forcing;             make)
make[1]: Entering directory '/home/pabecker/hrldas/hrldas/HRLDAS_forcing'
(cd lib; make)
make[2]: Entering directory '/home/pabecker/hrldas/hrldas/HRLDAS_forcing/lib'

rm -f module_geo_em.f90
cpp -P -traditional -D_GFORTRAN_  -DBIT32 -D_BZIP_NO module_geo_em.F > module_geo_em.f90
/home/linuxbrew/.linuxbrew/bin/gfortran -c -ffree-form  -ffree-line-length-none -g -fconvert=big-endian -fbounds-check -fno-range-check -fallow-argument-mismatch -I/home/linuxbrew/.linuxbrew/Cellar/netcdf-fortran/4.6.1/include module_geo_em.f90
rm -f libsmda.a
ar q libsmda.a module_grib2.o module_grib2_tables.o trig_degrees.o module_input_data_structure.o arguments_module.o gbytesys.o swap4f.o module_llxy.o cio.o kwm_date_utilities.o kwm_grid_utilities.o kwm_timing_utilities.o kwm_string_utilities.o swap4c.o get_unused_unit.o decode_jpeg2000.o io_f.o module_mapinfo.o module_grib1.o module_grib.o module_grib_common.o module_geo_em.o
ar: creating libsmda.a
make[2]: Leaving directory '/home/pabecker/hrldas/hrldas/HRLDAS_forcing/lib'
(make -f Makefile all)
make[2]: Entering directory '/home/pabecker/hrldas/hrldas/HRLDAS_forcing'

rm -f create_forcing.f90
cpp -P -traditional -D_GFORTRAN_   create_forcing.F > create_forcing.f90
/home/linuxbrew/.linuxbrew/bin/gfortran -ffree-form  -ffree-line-length-none -g -fconvert=big-endian -fbounds-check -fno-range-check -fallow-argument-mismatch -c -I/home/linuxbrew/.linuxbrew/Cellar/netcdf-fortran/4.6.1/include -I./lib -I./lib create_forcing.f90
(cd lib; make)
make[3]: Entering directory '/home/pabecker/hrldas/hrldas/HRLDAS_forcing/lib'
make[3]: Nothing to be done for 'all'.
make[3]: Leaving directory '/home/pabecker/hrldas/hrldas/HRLDAS_forcing/lib'
/home/linuxbrew/.linuxbrew/bin/gfortran -o create_forcing.exe -I./lib -g -fconvert=big-endian -fbounds-check -fno-range-check -fallow-argument-mismatch -I./lib create_forcing.o \
        -L./lib -lsmda -L/home/linuxbrew/.linuxbrew/Cellar/netcdf-fortran/4.6.1/lib -lnetcdf -lnetcdff  -L/home/linuxbrew/.linuxbrew/Cellar/jasper/4.2.4/lib
/usr/bin/ld: ./lib/libsmda.a(decode_jpeg2000.o): in function `info_jpeg2000_':
decode_jpeg2000.c:(.text+0x2d): undefined reference to `jas_init'
/usr/bin/ld: decode_jpeg2000.c:(.text+0x45): undefined reference to `jas_stream_memopen'
/usr/bin/ld: decode_jpeg2000.c:(.text+0x94): undefined reference to `jas_image_getfmt'
/usr/bin/ld: decode_jpeg2000.c:(.text+0xd9): undefined reference to `jpc_decode'
/usr/bin/ld: ./lib/libsmda.a(decode_jpeg2000.o): in function `decode_jpeg2000_':
decode_jpeg2000.c:(.text+0x1be): undefined reference to `jas_matrix_create'
/usr/bin/ld: decode_jpeg2000.c:(.text+0x221): undefined reference to `jas_image_readcmpt'
/usr/bin/ld: decode_jpeg2000.c:(.text+0x2bc): undefined reference to `jas_matrix_destroy'
/usr/bin/ld: decode_jpeg2000.c:(.text+0x2cb): undefined reference to `jas_stream_close'
/usr/bin/ld: decode_jpeg2000.c:(.text+0x2dd): undefined reference to `jas_image_destroy'
/usr/bin/ld: /home/linuxbrew/.linuxbrew/Cellar/netcdf-fortran/4.6.1/lib/libnetcdff.so: undefined reference to `nc_def_var_quantize'
/usr/bin/ld: /lib/x86_64-linux-gnu/libnetcdf.so: undefined reference to `curl_easy_getinfo@CURL_OPENSSL_4'
/usr/bin/ld: /home/linuxbrew/.linuxbrew/Cellar/netcdf-fortran/4.6.1/lib/libnetcdff.so: undefined reference to `nc_inq_var_quantize'
/usr/bin/ld: /lib/x86_64-linux-gnu/libnetcdf.so: undefined reference to `curl_global_init@CURL_OPENSSL_4'
/usr/bin/ld: /lib/x86_64-linux-gnu/libnetcdf.so: undefined reference to `curl_easy_perform@CURL_OPENSSL_4'
/usr/bin/ld: /lib/x86_64-linux-gnu/libhdf5_serial.so.103: undefined reference to `curl_slist_free_all@CURL_OPENSSL_4'
/usr/bin/ld: /lib/x86_64-linux-gnu/libnetcdf.so: undefined reference to `curl_easy_strerror@CURL_OPENSSL_4'
/usr/bin/ld: /lib/x86_64-linux-gnu/libnetcdf.so: undefined reference to `curl_easy_setopt@CURL_OPENSSL_4'
/usr/bin/ld: /lib/x86_64-linux-gnu/libnetcdf.so: undefined reference to `curl_easy_init@CURL_OPENSSL_4'
/usr/bin/ld: /lib/x86_64-linux-gnu/libhdf5_serial.so.103: undefined reference to `curl_slist_append@CURL_OPENSSL_4'
/usr/bin/ld: /lib/x86_64-linux-gnu/libnetcdf.so: undefined reference to `curl_easy_cleanup@CURL_OPENSSL_4'
/usr/bin/ld: /lib/x86_64-linux-gnu/libnetcdf.so: undefined reference to `curl_version_info@CURL_OPENSSL_4'
/usr/bin/ld: /lib/x86_64-linux-gnu/libnetcdf.so: undefined reference to `curl_global_cleanup@CURL_OPENSSL_4'
collect2: error: ld returned 1 exit status
make[2]: *** [Makefile:29: create_forcing.exe] Error 1
make[2]: Leaving directory '/home/pabecker/hrldas/hrldas/HRLDAS_forcing'
make[1]: *** [Makefile:13: default] Error 2
make[1]: Leaving directory '/home/pabecker/hrldas/hrldas/HRLDAS_forcing'
make: *** [Makefile:15: all] Error 2`

The hrldas.exe file is there under the run/ folder but the create_forcing.exe in the HRLDAS_forcing/ file is not

**Comment by ZhaoY-github** [comment]
Looks like your jasper lib is not correctly set. Did you install jasper on your machine and correctly linked in the **user_build_options** file?

**Reply by psbecker** [reply]
Got it thank you!
